engine-cli/src/main/java/io/nosqlbench/engine/cli/EBCLIOptions.java

```diff
@@ -313,10 +321,66 @@ private void parse(String[] args) {
                         Cmd script = parseScriptCmd(arglist);
                         cmdList.add(script);
                     } else {
-                        throw new InvalidParameterException("unrecognized option:" + word);
+                        Optional<Path> path = NosqlBenchFiles.findOptionalPath(word, "yaml", "activities", "activities/baselines");
+                        if(path.isPresent()){
+                            arglist.removeFirst();
+                            String scenarioFilter = null;
+                            if (arglist.size() > 0 && !arglist.peekFirst().contains("=")){
+                                scenarioFilter = arglist.peekFirst();
+                                arglist.removeFirst();
+                            };
+                            arglist.addFirst("yaml="+path.get().toString());
+                            parseWorkloadYamlCmds(path.get().toString(), arglist, scenarioFilter);
+                        }
+                        else {
+                            throw new InvalidParameterException("unrecognized option:" + word);
+                        }
                     }
+                    break;
+            }
+        }
+    }
+
+    private void parseWorkloadYamlCmds(String yamlPath, LinkedList<String> arglist, String scenarioFilter) {
+        StmtsDocList stmts = StatementsLoader.load(logger, yamlPath);
+
+        Scenarios scenarios = stmts.getDocScenarios();
+
+        String scenarioName = "default";
+        if (scenarioFilter != null){
+            scenarioName = scenarioFilter;
+        }
+
+        List<String> cmds = scenarios.getNamedScenario(scenarioName);
+
+        if (cmds == null){
+            List<String> names = scenarios.getScenarioNames();
+            throw new RuntimeException("Unknown scenaro name, make sure the scenario name you provide exists in the workload definition (yaml):\n" + String.join(",", names));
+        }
+
+        for (String cmd : cmds) {
+            String[] cmdArray = cmd.split(" ");
+
+            Map<String, String> paramMap = new HashMap<>();
+            for (String parameter: cmdArray) {
+                if (parameter.contains("=")){
+                    if ( !parameter.contains("TEMPLATE(") && !parameter.contains("<<")) {
+                        String[] paramArray = parameter.split("=");
+                        paramMap.put(paramArray[0], paramArray[1]);
+                    }
+                }
             }
+
+            StrSubstitutor sub1 = new StrSubstitutor(paramMap, "<<", ">>", '\\', ",");
+            StrSubstitutor sub2 = new StrSubstitutor(paramMap, "TEMPLATE(", ")", '\\', ",");
+
+            cmd = sub2.replace(sub1.replace(cmd));
+
+            // Is there a better way to do this than regex?
```

Comment:
I do think this will break if there is a case for spaces within parameter names or values, but otherwise, it should be ok.